app/adapters/application.js

```diff
@@ -1,3 +1,32 @@
 import JSONAPIAdapter from '@ember-data/adapter/json-api';
+import { service } from '@ember/service';
 
-export default class ApplicationAdapter extends JSONAPIAdapter {}
+const OFFENDING_STATUS_CODES = [401, 403];
+
+export default class ApplicationAdapter extends JSONAPIAdapter {
+  constructor() {
+    super(...arguments);
+    this.monkeyPatchFetch();
+  }
+
+  monkeyPatchFetch() {
+    const originalFetch = window.fetch;
+    const router = this.router;
+    window.fetch = async function () {
+      const response = await originalFetch.apply(this, arguments);
+      console.log('status', response.status);
+      if (OFFENDING_STATUS_CODES.indexOf(response.status) > -1) {
```

Comment:
Added a fetch to /something from the frontend repository and added the 304 status code to this array. It did not get triggered for some reason, maybe doing something wrong